test/integration_tests/test_select_executor.py

```diff
@@ -246,24 +245,120 @@ def test_select_and_sample(self):
+                        FastRCNNObjectDetector(data) AS T(a,b,c) WHERE id < 5;"""
         actual_batch = execute_query_fetch_all(select_query)
-        self.assertEqual(actual_batch.frames.columns, ["myvideo.id"])
+        self.assertEqual(list(actual_batch.columns), ["myvideo.id", "T.a"])
         self.assertEqual(actual_batch.batch_size, 5)
 
     @pytest.mark.torchtest
     def test_lateral_join_with_multiple_projects(self):
-        select_query = """SELECT id, labels FROM MyVideo JOIN LATERAL
-                        FastRCNNObjectDetector(data) WHERE id < 5;"""
+        select_query = """SELECT id, T.labels FROM MyVideo JOIN LATERAL
+                        FastRCNNObjectDetector(data) AS T WHERE id < 5;"""
         actual_batch = execute_query_fetch_all(select_query)
-        self.assertTrue(
-            all(
-                actual_batch.frames.columns
-                == ["myvideo.id", "fastrcnnobjectdetector.labels"]
+        self.assertTrue(all(actual_batch.frames.columns == ["myvideo.id", "T.labels"]))
+        self.assertEqual(actual_batch.batch_size, 5)
+
+    def test_lateral_join_with_unnest(self):
+        query = """SELECT id, label
+                  FROM MyVideo JOIN LATERAL
+                    UNNEST(DummyObjectDetector(data)) AS T(label)
+                  WHERE id < 2 ORDER BY id;"""
+        unnest_batch = execute_query_fetch_all(query)
+        expected = Batch(
+            pd.DataFrame(
+                {
+                    "myvideo.id": np.array([0, 1]),
+                    "T.label": np.array(["person", "bicycle"]),
+                }
             )
         )
-        self.assertEqual(actual_batch.batch_size, 5)
+
+        self.assertEqual(unnest_batch, expected)
+
+        query = """SELECT id, label
+                  FROM MyVideo JOIN LATERAL
+                    UNNEST(DummyObjectDetector(data)) AS T
+                  WHERE id < 2 ORDER BY id;"""
+        unnest_batch = execute_query_fetch_all(query)
+        expected = Batch(
+            pd.DataFrame(
+                {
+                    "myvideo.id": np.array([0, 1]),
+                    "T.label": np.array(["person", "bicycle"]),
+                }
+            )
+        )
+
+        self.assertEqual(unnest_batch, expected)
+
+    def test_lateral_join_with_unnest_on_subset_of_outputs(self):
+        query = """SELECT id, label
+                  FROM MyVideo JOIN LATERAL
+                    UNNEST(DummyMultiObjectDetector(data).labels) AS T(label)
+                  WHERE id < 2 ORDER BY id;"""
+        unnest_batch = execute_query_fetch_all(query)
+        expected = Batch(
+            pd.DataFrame(
+                {
+                    "myvideo.id": np.array([0, 0, 1, 1]),
+                    "T.label": np.array(["person", "person", "bicycle", "bicycle"]),
+                }
+            )
+        )
+
+        self.assertEqual(unnest_batch, expected)
+
+    def test_should_raise_error_with_missing_alias_in_lateral_join(self):
```

Comment:
So `JOIN LATERAL` much have `AS`. However, it seems from https://github.com/georgia-tech-db/eva/blob/3c16181b140bc3b8cd26e99212d73f14a479ba30/eva/binder/statement_binder.py#L281 , when the node does not have alias, it will use the default node name and column name.

Reply:
Yes, I have added the code to handle it. But later felt that avoiding it, for now, is better. We can later remove the check from the parser and test it more.